### PR TITLE
[SPARKTA-197] [DRIVER] Stop policies correctly

### PR DESCRIPTION
**Pull request by compae**
#### Description

Exceptions and error control when stopping spark context and actors.

#### Reviewer
@Gschiavon 

#### Test
All passed.

### CODE REVIEW

**Comment by Gschiavon** [comment]
+1

**Comment by anistal** [comment]
+1

**Comment by anistal** [comment]
When you try to stop a runned policy in a YARN environment, it stops correctly but instead of set the context node to "Stopped", it set the node to "Started".

File: driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala [hunk 1]
```diff
@@ -83,18 +84,35 @@ case class StreamingContextService(policyStatusActor: Option[ActorRef] = None, g
...
+            try {
+
+              log.info("Stopping message received from Zookeeper")
+              SparkContextFactory.destroySparkStreamingContext
+              SparkContextFactory.destroySparkContext
+
+            } finally {
+
+              Try(Await.result(policyStatusActor.get ? Kill(name), timeout.duration) match {
+                case false => log.warn(s"The actor with name: $name are stopped previously")
```
Comment (antonioalf) [inline driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala]: The actor with name: $name are stopped previously -> The actor with name: $name has been stopped previously

File: serving-core/src/main/scala/com/stratio/sparkta/serving/core/policy/status/PolicyStatusActor.scala [hunk 2]
```diff
@@ -40,22 +40,32 @@ class PolicyStatusActor(curatorFramework: CuratorFramework)
...
     pActor match {
-      case Some(Success(actor)) => context.stop(actor)
-      case Some(Failure(e)) => log.warn(s"there isn't $policyName actor to kill")
-      case None => log.warn(s"there isn't $policyName actor to kill")
+      case Some(Success(actor)) => {
+        context.stop(actor)
+        Stopped
+      }
+      case Some(Failure(e)) => {
+        log.warn(s"There isn't $policyName actor to kill")
```
Comment (antonioalf) [inline serving-core/src/main/scala/com/stratio/sparkta/serving/core/policy/status/PolicyStatusActor.scala]: There isn't $policyName actor to kill -> There is no policy with name: $policyName actor to kill

File: serving-core/src/main/scala/com/stratio/sparkta/serving/core/policy/status/PolicyStatusActor.scala [hunk 3]
```diff
@@ -40,22 +40,32 @@ class PolicyStatusActor(curatorFramework: CuratorFramework)
...
+        NotStopped
+      }
+      case None => {
+        log.warn(s"there isn't $policyName actor to kill")
```
Comment (antonioalf) [inline serving-core/src/main/scala/com/stratio/sparkta/serving/core/policy/status/PolicyStatusActor.scala]: There isn't $policyName actor to kill -> There is no policy actor with name: $policyName actor to kill

**Pull request merged**